src/labs/lit-extended.ts

```diff
@@ -44,16 +44,22 @@ import { TemplateResult, AttributePart, TemplateInstance, TemplatePart, Part, Te
  *
  */
 export function renderExtendedTo(result: TemplateResult, container: Element|DocumentFragment) {
-  let instance = container.__templateInstance as TemplateInstance;
-  if (instance === undefined) {
-    instance = new ExtendedTemplateInstance(result.template);
-    container.__templateInstance = instance;
-    const fragment = instance._clone();
-    instance.update(result.values);
-    container.appendChild(fragment);
-  } else {
+  let instance = container.__templateInstance as ExtendedTemplateInstance|TemplateInstance|undefined;
+  if (instance && instance.template === result.template) {
```

Comment:
also, explicitly check again `undefined`, it's faster because it skips a ToBoolean on `instance`